airflow/utils/decorators.py

```diff
@@ -21,13 +21,14 @@
 import os
 from copy import copy
 from functools import wraps
+from typing import Any, Callable, Dict
 
 from airflow.exceptions import AirflowException
 
 signature = inspect.signature
 
 
-def apply_defaults(func):
+def apply_defaults(func: Callable[..., Any]) -> Any:
```

Comment:
```suggestion
def apply_defaults(func: Callable[..., Any]) -> Callable[..., Any]:
```

Reply:
This is a bigger challenge. mypy is not too smart to handle this decorator correctly. Please look at: 
https://github.com/apache/airflow/pull/8145
https://github.com/apache/airflow/tree/master/airflow/mypy
https://github.com/python/mypy/issues/3157